client/src/widgets/qdateeditex.cpp

```diff
@@ -21,61 +21,72 @@
 **************************************************************************/
 
 #include "qdateeditex.h"
-#include <QStyle>
-#include <QPushButton>
 #include <QLineEdit>
-#include <QStyleOptionSpinBox>
 #include <QKeyEvent>
-#include <QCalendarWidget>
+#include <QAbstractItemView>
+#include <QAbstractButton>
+#include <QDate>
+#include <QComboBox>
+#include <QMenu>
+
 #include "fatcrm_client_debug.h"
 
 /*!
   \class DateEditEx dateeditex.h
   \brief A DateEdit with a nullable date
 
-  This is a subclass of QDateEdit that has the additional feature
+  This is a subclass of KDateComboBox that has the additional feature
   of allowing to select a empty date. This can be achived with the
   clear button or by selecting the whole date and press backspace.
 
   To set an empty date from code, use the setter with an invalid
   date. To check whether the date is empty check if the date is valid,
   which comes from the getter.
 
-  \sa QDateEdit
+  \sa KDateComboBox
 */
 
 class QDateEditEx::Private {
 public:
-    Private( QDateEditEx* qq ) : q(qq),  clearButton(nullptr), null(false), nullable(false) {}
+    Private( QDateEditEx* qq ) : q(qq), null(false), nullable(false) {}
 
     QDateEditEx* const q;
 
-    QPushButton* clearButton;
-
     bool null;
     bool nullable;
 
     void setNull(bool n) {
         null = n;
         if (null) {
-            QLineEdit *edit = qFindChild<QLineEdit *>(q, QStringLiteral("qt_spinbox_lineedit"));
+            QLineEdit *edit = q->lineEdit();
             if (!edit->text().isEmpty()) {
                 edit->clear();
             }
         }
-        if (nullable) {
-            clearButton->setVisible(!null);
-        }
 
+        q->lineEdit()->setClearButtonEnabled(nullable);
     }
 };
 
 /*!
   \reimp
 */
 QDateEditEx::QDateEditEx(QWidget *parent) :
-    QDateEdit(parent), d(new Private(this))
+    KDateComboBox(parent), d(new Private(this))
 {
+    // set object names for all internal children to avoid it to be hidden by Details widget
```

Comment:
s/it/them/   (children is plural)
s/to be/being/   (avoid is followed by a noun)